packages/router/src/typed-routes/route-map.ts

```diff
@@ -18,13 +18,15 @@ export interface RouteRecordInfo<
   ParamsRaw extends RouteParamsRawGeneric = RouteParamsRawGeneric,
   Params extends RouteParamsGeneric = RouteParamsGeneric,
   Meta extends RouteMeta = RouteMeta,
+  ChildrenNames extends string | symbol | never = never | string | symbol,
```

Comment:
Still figuring out a way to keep a RouteRecordInfo with type params assignable to the generic one (no type params).